### PR TITLE
Add environment commands

### PR DESCRIPTION
**Pull request by eugeneantonevich**

### CODE REVIEW

File: src/common/context.ts [hunk 1]
```diff
@@ -80,17 +81,37 @@ export class Context {
     return null;
   }
 
+  async getEnvironments(workspaceId: string): Promise<EnvironmentInfo[]> {
+    const { system } = await this.request(GraphqlActions.environmentsList, null, true, workspaceId, false);
+
+    const environments = system.environments.items;
+    if (_.isEmpty(environments)) {
+      throw new Error(this.i18n.t('logout_error'));
```
Comment (zouxuoz) [inline src/common/context.ts:89]: @eugeneantonevich why logout error?

Comment (eugeneantonevich) [reply src/common/context.ts:89]: the same as getWorkspaces function

**Comment by eugeneantonevich** [comment]
TODO: we have to review and fix help message

File: src/engine/commands/backup/commands/export.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  command: 'export',
+  handler: async (params: any, context: Context) => {
+    ProjectConfigurationState.expectConfigured(context);
+    const { environment, backup } = params;
+    const {
+      system: {
+        environmentBackupUrl: { url },
+      },
+    } = await context.request(GraphqlActions.backupUrl, { name: environment, backup });
+    console.log(url);
```
Comment (esemushin) [inline src/engine/commands/backup/commands/export.ts:17]: @eugeneantonevich

File: src/consts/GraphqlActions.ts [hunk 3]
```diff
@@ -83,4 +106,40 @@ export const GraphqlActions = {
...
+
+export const GraphqlAsyncActions = {
+  environmentBranch: `
+    mutation clone($environmentName: String!, $mode: SystemBranchEnvironmentMode) {
+      system { async: environmentBranch(name: $environmentName mode: $mode) { sessionId } }
+    }`,
+  commit: `
+    mutation CommitMigration($mode: SystemCiCommitMode! $build:String) {
+      system { async: ciCommit(mode:$mode build:$build) { sessionId } }
+    }
```
Comment (ej9x) [inline src/consts/GraphqlActions.ts:123]: @eugeneantonevich Maybe either write commas separator in all queries or not?

File: src/locales/en.ts [hunk 4]
```diff
@@ -276,6 +277,138 @@ export default {
...
+    environment_set_describe: 'Set environment',
+    environment_set_usage: '8base environment set',
+    environment_set_environment_name_describe: 'The environment name of the project',
+
+    migration_describe: 'Migration commands.',
+
+    /**
+     * Migration commit related messages
+     */
+
```
Comment (ej9x) [inline src/locales/en.ts:361]:
```suggestion
```

File: src/locales/en.ts [hunk 5]
```diff
@@ -276,6 +277,138 @@ export default {
...
+    migration_status_in_progress: 'Resolve status...',
+    migration_status_environment_describe: 'Target environment',
+
+    /**
+     * Migration status related messages
+     */
+    environment_show_usage: 'COMMAND\n  8base environment show\n\nDESCRIPTION\n Display current environment.',
+    environment_show_describe: '',
+    environment_show_text: 'Environment: {{-environment}}',
+    /**
```
Comment (ej9x) [inline src/locales/en.ts:402]:
```suggestion

    /**
```

File: src/locales/en.ts [hunk 6]
```diff
@@ -276,6 +277,138 @@ export default {
...
+     * Migration apply related messages
+     */
+    migration_apply_in_progress: 'Migration apply in progress',
+    migration_apply_describe: 'Migration apply',
+
+    /**
+     * Plugin related messages
+     */
+
```
Comment (ej9x) [inline src/locales/en.ts:411]:
```suggestion
```

File: src/common/context.ts [hunk 7]
```diff
@@ -211,12 +234,19 @@ export class Context {
...
   }
 
-  async request(
-    query: string,
-    variables: any = null,
-    isLoginRequired = true,
-    customWorkspaceId?: string,
-  ): Promise<any> {
+  async request(query: string, variables: any = null, options?: RequestOptions): Promise<any> {
+    const DefaultOptions: RequestOptions = {
```
[the remark this answers is not in the document]
Comment (eugeneantonevich) [reply src/common/context.ts:238]: yep

**Pull request merged**